distributed/scheduler.py

```diff
@@ -3497,6 +3502,24 @@ def remove_all_replicas(self, ts: TaskState):
             self._replicated_tasks.remove(ts)
         ts._who_has.clear()
 
+    @ccall
+    @exceptval(check=False)
+    def _reevaluate_occupancy_worker(self, ws: WorkerState):
+        """See reevaluate_occupancy"""
+        ts: TaskState
+        old = ws._occupancy
+        for ts in ws._processing:
+            self.set_duration_estimate(ts, ws)
+
+        self.check_idle_saturated(ws)
+        steal = self.extensions.get("stealing")
+        if steal is None:
+            return
+        if ws._occupancy > old * 1.3 or old > ws._occupancy * 1.3:
+            for ts in ws._processing:
+                steal.remove_key_from_stealable(ts)
+                steal.put_key_in_stealable(ts)
```

Comment:
I assume this is to trigger re-evaluation of stealing, but it looks kind of odd reading it. Comment?

Reply:
yes, there is now a "better" API on the stealing extension which is called recompute cost or smth like that which does this in the background. the api of put/remove has been abused in the past a bit and I should change this

Reply:
- [ ] Use `steal.recalculate_cost` API